File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        if nums == []: return False
        nums.sort()
        last_n = nums[0]
        for n in nums[1:]:
            if last_n == n:
                return True
            last_n = n
        return False

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> counter(2001);
        for (int n: nums) {
            counter[n + 1000]++;
        }

        priority_queue<pair<int, int>> maxheap;
        for (int i = 0; i < 2001; i++) {
            maxheap.push({counter[i], i - 1000});
        }
        
        vector<int> result(k);
        for (int i = 0; i < k; i++) {
            auto [c, n] = maxheap.top();
            result[i] = n;
            maxheap.pop();
        }

        return result;
    }
};
